src/platform/test.rs

```diff
@@ -414,6 +414,54 @@ fn receiver_set() {
 }
 
 #[test]
+#[cfg(not(any(feature = "force-inprocess", target_os = "windows", target_os = "android")))]
+fn receiver_set_eintr() {
+    let (tx0, rx0) = platform::channel().unwrap();
+    let (tx1, rx1) = platform::channel().unwrap();
+    let mut rx_set = OsIpcReceiverSet::new().unwrap();
+    rx_set.add(rx1).unwrap();
+    let child_pid = unsafe {
+        fork(|| {
+            // Let the parent know we're ready
+            tx0.send(b"\x00", vec![], vec![]).unwrap();
+            // Send the result of the select back to the parent
+            match rx_set.select() {
+                Ok(result) => {
+                    for evt in result {
+                        if let OsIpcSelectionResult::DataReceived(_, data, _, _) = evt {
+                            assert!(data == b"\x00");
```

Comment:
I'm sorry to say that the error checking is both overly complicated and incomplete: it doesn't check for wrong number of results (including 0); nor for unexpected `ChannelClosed` instead of `DataReceived`...

As I mentioned before, you can just do `let (id, data, _, _) = result.into_iter().next().unwrap().unwrap()`, like existing test cases.

(In case it's not clear, the first unwrap() is for the `next()` returning `Some(...)`, while the second one is a custom method of `OsIpcSelectionResult`.)

Or being really thorough:
```rust
let mut result_iter = result.into_iter();
let (id, data, _, _) = result_iter.next().unwrap().unwrap();
assert!(id == rx_id);
assert!(data == b"\x00");
assert!(result_iter.next().is_none());
```